core/services/src/main/java/org/exoplatform/services/rest/AttachmentsRestService.java

```diff
@@ -302,6 +301,39 @@ public Response downloadActivityAttachments(
     return Response.seeOther(location).build();
   }
 
+  @POST
+  @Path("/{attachmentId}/move")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Move an attachment to a destination path", httpMethod = "POST", response = Response.class, consumes = "application/json", notes = "returns empty response")
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.NO_CONTENT, message = "Request fulfilled"),
+      @ApiResponse(code = HTTPStatus.BAD_REQUEST, message = "Invalid query input"),
+      @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+      @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error") })
+  public Response moveAttachmentToNewPath(@ApiParam(value = "New path", required = true) @QueryParam("newPath") String newPath,
+                                          @ApiParam(value = "New destination path's drive", required = true) @QueryParam("newPathDrive") String newPathDrive,
+                                          @ApiParam(value = "New destination path's drive", required = true) @QueryParam("entityType") String entityType,
+                                          @ApiParam(value = "Entity technical identifier", required = true) @QueryParam("entityId") long entityId,
+                                          @ApiParam(value = "Entity type", required = true) @PathParam("attachmentId") String attachmentId) {
+
+    if (StringUtils.isEmpty(newPathDrive)) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("New destination path's drive is mandatory").build();
+    }
+
+    if (StringUtils.isEmpty(attachmentId)) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Attachment id is mandatory").build();
+    }
+
+    long userIdentityId = getCurrentUserIdentityId();
+    try {
+      attachmentService.moveAttachmentToNewPath(userIdentityId, attachmentId, newPathDrive, newPath, entityType, entityId);
+    } catch (Exception e) {
+      LOG.error("Error when trying to move attachment with id {} to new destination path {} ", attachmentId, newPath, e);
+      return Response.serverError().entity(e.getMessage()).build();
```

Comment:
it's better to avoid giving the user access to the exception message which can hold a sensitive data for logging only.
You should add a simple message that can helps the user to change his query, or nothing when it's about a generic exception as here.